### PR TITLE
NXP-28802: Make fixed part of document history pp configurable

### PR DESCRIPTION
**Pull request by michaelva**

### CODE REVIEW

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/main/java/org/nuxeo/elasticsearch/audit/pageprovider/ESDocumentHistoryPageProvider.java [hunk 1]
```diff
@@ -79,6 +79,14 @@
 
     @Override
     protected String getFixedPart() {
+        if (getDefinition().getWhereClause() != null) {
+            //if the pp definition contains a fixed part, use it
+            String fixedPart = getDefinition().getWhereClause().getFixedPart();
+            if (fixedPart != null && !fixedPart.isEmpty()) {
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/main/java/org/nuxeo/elasticsearch/audit/pageprovider/ESDocumentHistoryPageProvider.java:85]:
```suggestion
            if (StringUtils.isNotEmpty(fixedPart)) {
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 2]
```diff
@@ -197,6 +200,9 @@ protected void createTestEntries() throws Exception {
         createdEntry.setEventDate(new Date());
         createdEntry.setDocPath(doc.getPathAsString());
         createdEntry.setRepositoryId("test");
+        Map<String, ExtendedInfo> extendedInfos = new HashMap<>();
+        extendedInfos.put("reason",new ExtendedInfoImpl.StringInfo("test"));
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:204]:
```suggestion
        extendedInfos.put("reason", new ExtendedInfoImpl.StringInfo("test"));
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 3]
```diff
@@ -398,6 +404,30 @@ public void testCustomDocumentHistoryPageProvider() {
...
     }
 
+    @Test
+    public void testCustomFixedPartDocumentHistoryPageProvider() {
+        assertNotNull(pageProviderService.getPageProviderDefinition("FIXED_PART_DOCUMENT_HISTORY_PROVIDER"));
+        DocumentModel searchDoc = session.createDocumentModel("BasicAuditSearch");
+        searchDoc.setPathInfo("/", "auditsearch");
+        searchDoc = session.createDocument(searchDoc);
+
+        PageProvider<LogEntry> pageProvider;
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:414]: Move it to the first `pageProvider` assignation.

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 4]
```diff
@@ -398,6 +404,30 @@ public void testCustomDocumentHistoryPageProvider() {
...
+        searchDoc.setPropertyValue("basicauditsearch:eventIds", null);
+        searchDoc.setPropertyValue("basicauditsearch:eventCategories", null);
+        searchDoc.setPropertyValue("basicauditsearch:startDate", null);
+        searchDoc.setPropertyValue("basicauditsearch:endDate", null);
+
+        //test with doc
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:420]:
```suggestion
        // test with doc
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 5]
```diff
@@ -398,6 +404,30 @@ public void testCustomDocumentHistoryPageProvider() {
...
+        pageProvider = getPageProvider("FIXED_PART_DOCUMENT_HISTORY_PROVIDER", 30, 0, doc);
+        pageProvider.setSearchDocumentModel(searchDoc);
+        assertEquals(1,pageProvider.getCurrentPage().size());
+
+        //test with proxy to check that the doc uuid is correctly set in the fixed part
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:425]:
```suggestion
        // test with proxy to check that the doc uuid is correctly set in the fixed part
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 6]
```diff
@@ -398,6 +404,30 @@ public void testCustomDocumentHistoryPageProvider() {
...
+        PageProvider<LogEntry> pageProvider;
+        searchDoc.setPropertyValue("basicauditsearch:eventIds", null);
+        searchDoc.setPropertyValue("basicauditsearch:eventCategories", null);
+        searchDoc.setPropertyValue("basicauditsearch:startDate", null);
+        searchDoc.setPropertyValue("basicauditsearch:endDate", null);
+
+        //test with doc
+        pageProvider = getPageProvider("FIXED_PART_DOCUMENT_HISTORY_PROVIDER", 30, 0, doc);
+        pageProvider.setSearchDocumentModel(searchDoc);
+        assertEquals(1,pageProvider.getCurrentPage().size());
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:423]:
```suggestion
        assertEquals(1, pageProvider.getCurrentPage().size());
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java [hunk 7]
```diff
@@ -398,6 +404,30 @@ public void testCustomDocumentHistoryPageProvider() {
...
+
+        //test with proxy to check that the doc uuid is correctly set in the fixed part
+        pageProvider = getPageProvider("FIXED_PART_DOCUMENT_HISTORY_PROVIDER", 30, 0, proxy);
+        pageProvider.setSearchDocumentModel(searchDoc);
+        assertEquals(0,pageProvider.getCurrentPage().size());
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java:428]:
```suggestion
        assertTrue(pageProvider.getCurrentPage().isEmpty());
```

File: modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/resources/es-audit-pageprovider-test-contrib.xml [hunk 8]
```diff
@@ -128,6 +128,45 @@
...
+          <field schema="basicauditsearch" name="principalNames" />
+        </predicate>
+      </whereClause>
+      <sort column="eventDate" ascending="false" />
+      <pageSize>10</pageSize>
+      <maxPageSize>1000</maxPageSize>
+    </genericPageProvider>
+
+
+
```
Comment (troger) [inline modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/resources/es-audit-pageprovider-test-contrib.xml:169]: Not really needed 😃

**Comment by michaelva** [comment]
@troger thanks for the review. I updated the PR with the suggestions you made.

**Comment by michaelva** [comment]
sorry, need one more approval to be able to merge

**Comment by kevinleturc** [comment]
Do a T&P has been run?

**Comment by michaelva** [comment]
@kevinleturc I thought it happens automatically. Isn't it all the utest/ftests checks in the pull request?

**Comment by kevinleturc** [comment]
Not yet, these statuses are still a work in progress, they become more and more reliable and we will (soon) leverage them. At this moment, we will set up these statuses as required on GitHub.

I started a T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-kleturc-2/199

**Comment by kevinleturc** [comment]
Merged by jenkins.

**Pull request closed without merging**